src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/test_webapp_commands.py

```diff
@@ -192,6 +192,27 @@ def test_download_linux_web_log(self, resource_group):
         self.assertTrue(os.path.isdir(os.path.join(log_dir, 'LogFiles', 'kudu', 'trace')))
 
 
+class WebappZipDeployScenarioTest(ResourceGroupVCRTestBase):
+    def __init__(self, test_method):
+        super(WebappZipDeployScenarioTest, self).__init__(__file__, test_method, resource_group='cli-webapp-zipDeploy')
+        self.webapp_name = 'azurecli-webapp-zipdeploy'
+
+    def test_deploy_zip(self):
+        self.execute()
+
+    def body(self):
+        plan = 'webapp-zip-deploy-plan'
+        zip_file = os.path.join(TEST_DIR, 'test.zip')
+        self.cmd('appservice plan create -g {} -n {} --sku B1'.format(self.resource_group, plan))
+        self.cmd('webapp create -g {} -n {} --plan {}'.format(self.resource_group, self.webapp_name, plan))
+        self.cmd('webapp deployment source config-zip -g {} -n {} --src "{}"'.format(self.resource_group, self.webapp_name, zip_file))
+        # wait for deployment to finish before verification
+        time.sleep(30)
```

Comment:
Please use polling instead of sleep for an arbitrary period of time. This make test unstable for the live environment.

Reply:
@github@troydai.com - i see sleep used across multiple tests in this file. I am unclear how is this different from the other? Please refer to the test right above.

---
In reply to: [145751592](https://github.com/Azure/azure-cli/pull/4700#discussion_r145751592) [](ancestors = 145751592)

Reply:
Yes, but those are bad examples. We didn't capture those failures back then and we will gradually fix them. We didn't run automation in the live environment often before, but we're doing this more today. Sleep in the code not only slow down the live automation but also introduce uncertainty when validating test error.

Simply put the code request this test into a loop, sleep for short period of time in each loop and tolerant and retry against failures for the limited number of times. This makes test code more responsiveness.